pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -245,12 +245,17 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 	}
 	c.networkManager = initNetworkManager(c, options)
 
-	c.namespaces = kclient.NewFiltered[*v1.Namespace](kubeClient, kclient.Filter{ObjectFilter: kubeClient.ObjectFilter()})
+	// currently NOT using kubeClient.ObjectFilter() here as we only care about the system namespace
+	// if in the future we want to watch other namespaces here, we will need to alter the discoveryNamespacesFilter
+	// to have an exception for the system namespace
+	c.systemNamespace = kclient.NewFiltered[*v1.Namespace](kubeClient, kclient.Filter{
+		LabelSelector: "kubernetes.io/metadata.name=" + c.opts.SystemNamespace,
```

Comment:
I would remove this, it actually will add more load on the API server since we are now opening 2 watches instead of 1. We already have the client side filter on L256 (or we can add it here too if we need it)